### PR TITLE
[WIP] UTF8 by the book

### PR DESCRIPTION
**Pull request by noirbizarre**
This PR tries to properly handle utf-8 on edge cases.

Detection:
- forces systeme default encoding to `ascii` during test
- makes the factories use unicode when using lorem generator
- feeds factories with more data (use traits)
- try to test facets rendering

Resolution:
- fixes the `__str__` and `__unicode__` handler on some models

### CODE REVIEW

File: udata/utils.py [hunk 1]
```diff
@@ -230,3 +237,9 @@ class UDataProvider(BaseProvider):
     def unique_string(self, length=UUID_LENGTH):
         '''Generate a unique string'''
         return unique_string(length)
+
+
+@faker_provider  # Replace the default lorem provider with a unicode one
+class UnicodeLoremProvider(LoremProvider):
+    '''A Lorem provider that forces unicode in words'''
```
Comment (abulte) [inline udata/utils.py]: 👍
